### PR TITLE
Clarify lock usage

### PR DESCRIPTION
**Pull request by toshipp**
We use some mutexes, but it is unclear what for. This PR adds comments or refactors code to improve meanings.
The most notable part is moving a locking functionality from LogicalVolumeService to Node/ControllerServer. Also, note that the new lock type `LockByID` improves parallelism.

### CODE REVIEW

**Review by llamerada-jp — CHANGES_REQUESTED** [review]

File: driver/lock_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		close(c)
+	}()
+
+	select {
+	case <-c:
+		t.Error("Failed to prevent concurrent execution")
+	case <-time.After(time.Second):
+		// success
+	}
+}
```
[the remark this answers is not in the document]
Comment (toshipp) [reply driver/lock_test.go:27]: Do you mean adding a test case to check `c` is closed properly after `UnlockByID`?

[the remark this answers is not in the document]
Comment (llamerada-jp) [reply driver/lock_test.go:27]: Yes, I do.

File: driver/node.go [hunk 2]
```diff
@@ -21,48 +21,103 @@ import (
...
+			k8sLVService: lvService,
+			mounter: mountutil.SafeFormatAndMount{
+				Interface: mountutil.New(""),
+				Exec:      utilexec.New(),
+			},
+		},
+	}, nil
+}
+
+// This is a wrapper for nodeServerNoLocked to protect concurrent method call.
```
Comment (cupnes) [inline driver/node.go:58]:
```suggestion
// This is a wrapper for nodeServerNoLocked to protect concurrent method calls.
```

File: driver/node.go [hunk 3]
```diff
@@ -21,48 +21,103 @@ import (
...
+type nodeServer struct {
+	csi.UnimplementedNodeServer
+
+	// This protects concurrent nodeServerNoLocked method calls.
+	// We use a global lock because it assumes that each method does not take long time,
```
Comment (cupnes) [inline driver/node.go:63]:
```suggestion
	// We use a global lock because it assumes that each method does not take a long time,
```

**Review by cupnes — COMMENTED** [review]
Please some small fixes if you need them.

**Review by llamerada-jp — APPROVED** [review]

**Review by cupnes — APPROVED** [review]

**Pull request merged**